pkg/universe.dagger.io/CODING_STYLE.md

```diff
@@ -0,0 +1,203 @@
+# Package Coding Style
+
+## Public names
+
+Choose `PascalCase` for ”public“ definitions (importable outside of package).
+
+```cue
+// good
+#WriteFile: {}
+
+// bad
+#writeFile: {}
+#write_file: {}
+```
+
+Choose `camelCase` for "public" fields, instead of `snake_case` or `"kebab-case"`.
+
+```cue
+actions: {
+    // good
+    goLint: ...
+
+    // bad
+    go_lint: ...
+    "go-lint": ...
+}
+```
+
+## Private names
+
+Choose `_#camelCase` for private definitions, that should only be used in current package.
+
+```cue
+// good
+_#mergeStructs: { ... }
+
+// bad, starts with lower case but not private
+#mergeStructs: { ... }
+
+// bad, starts with upper case but private
+_#MergeStructs: { ... }
+```
+
+Choose `_camelCase` for private fields (implementation details).
+
+```cue
+#Copy: {
+    // good, indicates internal implementation
+    _copy: core.#Copy
+
+    // bad, indicates it can be used directly
+    copy: core.#Copy
+}
+```
+
+## Definitions (`#`) for *schemas*, fields for concrete *implementations*
+
+```cue
+// good, defines a schema
+#Copy: {
+    // good, implements a schema
+    input: #Image
+    _copy: core.#Copy & { ... }
+}
+
+// bad, not a definition
+Copy: {
+    // bad, not using a visible or hidden field (according to intent)
+    #input: #Image
+    _#copy: core.#Copy & { ... }
+}
+
+// ok for mixin
+Mixin: { ... }
+copy: #Copy & Mixin & { ... }
+```
+
+## All public definitions and fields must be documented
+
+Other fields are optional. The main idea is that public fields should be documented for someone
+reading Dagger’s documentation in the browser. Everything else is for someone looking at the code.
+
+```yaml
+// Copy files into an image
+#Copy: {
+    // Image to receive copied files
+    input: #Image
+
+    // Files to copy
+    contents: dagger.#FS
+
+    _copy: ...
+
+    // Resulting image with added files
+    output: ...
+}
+```
+
+## Use verbs for action names, nouns otherwise
+
+Action definition names should be verbs (e.g., `#Build`, `#Run`, `#Push`),
+otherwise they should be nouns as much as possible (e.g., `#Image`, `#Secret`, `#Socket`).
```

Comment:
The first part of the sentence make sense to me. The second part "otherwise they should be nouns" is confusing, because "they" seems to refer to "action definition names".

Reply:
Indeed 🙂